stdlib/io.pyi

```diff
@@ -94,7 +94,10 @@ class BufferedIOBase(IOBase):
 
 class FileIO(RawIOBase, BinaryIO):  # type: ignore[misc]  # incompatible definitions of writelines in the base classes
     mode: str
-    name: FileDescriptorOrPath
+    # The type of "name" equals the argument passed in to the constructor,
+    # but that can make FileIO incompatible with other I/O types that assume
+    # "name" is a str. In the future, making FileIO generic might help.
+    name: Any
```

Comment:
Are you sure it's `FileIO.name` that has the incorrect annotation, and not `_WrappedBuffer.name`? Does `_WrappedBuffer.name` _have_ to be a `str`?

Note that `TextIOWrapper.name` just returns the `name` attribute of the underlying buffer, and our annotations for `TextIOWrapper` currently have it just inheriting the `name` property from `typing.TextIO`, meaning that type checkers currently infer `TextIOWrapper.name` to be a property return `str | Any`:

https://github.com/python/typeshed/blob/a3d356c04c15afc22b254afae2e6e7adbbe031a5/stdlib/typing.pyi#L708-L711

So it's a bit weird for us to say "the underlying buffer object has to have a `name` attribute that's a `str`, but the `TextIOWrapper` object could return anything from its `name` property"

Reply:
The I/O classes are just hilariously incoherent. I'm fine with just using `Any` in the protocol and/or restoring the `FileIO.name` annotation, although leaving the latter as `Any` for now is probably not a bad idea either.

Reply:
My weak preference would be to use `Any` (or possibly `Incomplete`) for the new protocol for now, as it seems least likely to lead to unexpected and unintended false positives